android/api.md

```diff
@@ -594,6 +609,252 @@ private static String MyOIDCAuthProvider(){
+
+```
+{
+  "Version": "1.0",
+  "AppSync": {
+    "Default": {
+      "ApiUrl": "https://xyz.us-west-2.amazonaws.com/graphql",
+      "Region": "us-west-2",
+      "AuthMode": "API_KEY",
+      "ApiKey": "da2-xyz",
+      "ClientDatabasePrefix": "friendly_name_API_KEY"
+    },
+    "friendly_name_AWS_IAM": {
+      "ApiUrl": "https://xyz.us-west-2.amazonaws.com/graphql",
+      "Region": "us-west-2",
+      "AuthMode": "AWS_IAM",
+      "ClientDatabasePrefix": "friendly_name_AWS_IAM"
+    },
+    "friendly_name_AMAZON_COGNITO_USER_POOLS": {
+      "ApiUrl": "https://xyz.us-west-2.amazonaws.com/graphql",
+      "Region": "us-west-2",
+      "AuthMode": "AMAZON_COGNITO_USER_POOLS",
+      "ClientDatabasePrefix": "friendly_name_AMAZON_COGNITO_USER_POOLS"
+    },
+    "friendly_name_OPENID_CONNECT": {
+      "ApiUrl": "https://xyz.us-west-2.amazonaws.com/graphql",
+      "Region": "us-west-2",
+      "AuthMode": "OPENID_CONNECT",
+      "ClientDatabasePrefix": "friendly_name_OPENID_CONNECT"
+    }
+  }
+}
+```
+
+The `useClientDatabasePrefix` is added on the client builder which signals to the builder that the `ClientDatabasePrefix` should be used from the AWSConfiguration object (*awsconfiguration.json*).
+
+```java
+AWSAppSyncClient client = AWSAppSyncClient.builder()
+                               .context(getApplicationContext())
+                               .awsConfiguration(new AWSConfiguration(getApplicationContext()))
+                               .useClientDatabasePrefix(true)
+                               .build();
+```
+
+The following code creates a client factory to retrieve the client based on the authorization mode: public (API Key) or private (AWS IAM).
+
+```java
+// AppSyncClientMode.java
+public enum AppSyncClientMode {
+    PUBLIC,
+    PRIVATE
+}
+
+// ClientFactory.java
+public class ClientFactory {
+
+    private static Map<AWSAppSyncClient> CLIENTS;
+
+    public static AWSAppSyncClient getAppSyncClient(AppSyncClientMode choice) {
+        return CLIENTS[choice];
+    }
+
+    public static void initClients(final Context context) {
+        AWSConfiguration awsConfigPublic = new AWSConfiguration(context);
+        CLIENTS[PUBLIC] = AWSAppSyncClient.builder()
+                                  .context(context)
+                                  .awsConfiguration(awsConfigPublic)
+                                  .useClientDatabasePrefix(true)
+                                  .build();
+        
+        AWSConfiguration awsConfigPrivate = new AWSConfiguration(context);
+        awsConfigPrivate.setConfiguration("friendly_name_AWS_IAM");
+        CLIENTS[PRIVATE] = AWSAppSyncClient.builder()
+                                   .context(context)
+                                   .awsConfiguration(awsConfigPrivate)
+                                   .useClientDatabasePrefix(true)
+                                   .cognitoUserPoolsAuthProvider(new CognitoUserPoolsAuthProvider() {
```

Comment:
This should show the `.credentialsProvider()` call not userpools.

Reply:
Fixed.